docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,401 @@
+=== Client dependencies and implementation
+
+Also it is needed to include the following dependencies for the client application: 
+
+[[client-dependencies]]
+=== Client dependencies
+Example:
+[source,XML]
+----
+    <dependency>
+        <groupId>io.grpc</groupId>
+        <artifactId>grpc-netty-shaded</artifactId>
+        <version>1.43.1</version>
+        <scope>runtime</scope>
+    </dependency>
+    <dependency>
+        <groupId>io.grpc</groupId>
+        <artifactId>grpc-protobuf</artifactId>
+        <version>1.43.1</version>
+    </dependency>
+    <dependency>
+        <groupId>io.grpc</groupId>
+        <artifactId>grpc-stub</artifactId>
+        <version>1.43.1</version>
+    </dependency>
+    <dependency> <!-- necessary for Java 9+ -->
+        <groupId>org.apache.tomcat</groupId>
+        <artifactId>annotations-api</artifactId>
+        <version>6.0.53</version>
+        <scope>runtime</scope>
+    </dependency>
+----
+
+And the following is the client code to consume the service
+
+[[client-code-implementation]]
+=== Client Code implementation
+[source, Java]
+----
+public class GrpcClient {
+    private static final Logger LOGGER = Logger.getLogger(GrpcClient.class.getName());
+    private final PayaraServiceGrpc.PayaraServiceStub stub;
+    private CountDownLatch latch;
+    private AtomicReference<Throwable> error;
+
+    public static void main(String[] args) throws InterruptedException, MalformedURLException, URISyntaxException {
+        URL myURL = new URL("http://localhost:8080/fish.payara.samples.grpc.PayaraService");
+        final GrpcClient client = new GrpcClient(myURL);
+        client.communicate();
+    }
+
+    public GrpcClient(URL url) throws URISyntaxException {
+        final Channel channel = ManagedChannelBuilder.forAddress(url.getHost(), url.getPort())
+                .usePlaintext().build();
+        this.stub = PayaraServiceGrpc.newStub(channel);
+        this.error = new AtomicReference<>(null);
+    }
+
+    public void communicate() throws InterruptedException {
+        latch = new CountDownLatch(1);
+        stub.communicate(request("Hello World"), new ResponseObserver());
+        latch.await(20, TimeUnit.SECONDS);
+    }
+
+    public Throwable getError() {
+        return error.get();
+    }
+
+    private final class ResponseObserver implements StreamObserver<PayaraResp> {
+
+        @Override
+        public void onNext(PayaraResp response) {
+            LOGGER.log(Level.INFO, "Response received: \"{0}\".", response.getMessage());
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            LOGGER.log(Level.SEVERE, "Error received", t);
+            error.set(t);
+            latch.countDown();
+        }
+
+        @Override
+        public void onCompleted() {
+            latch.countDown();
+        }
+
+    }
+
+    private static final PayaraReq request(String message) {
+        return PayaraReq.newBuilder().setMessage(message).build();
+    }
+}
```

Comment:
Might be good to reference external documentation so that users have more information about the requirements needed to code properly their client applications since there are no details at all.